lib/openstax/aws/auto_scaling_instance.rb

```diff
@@ -0,0 +1,97 @@
+module OpenStax::Aws
+  class AutoScalingInstance
+
+    attr_reader :raw
+
+    delegate_missing_to :@raw
+
+    def initialize(group_name:, id:, region:)
+      @raw = Aws::AutoScaling::Instance.new(
+        group_name,
+        id,
+        client: Aws::AutoScaling::Client.new(region: region)
+      )
+    end
+
+    def self.me
+      instance_id = Ec2InstanceData.instance_id
+      region = Ec2InstanceData.region
+
+      client = Aws::AutoScaling::Client.new(region: region)
+      instance_info = client.describe_auto_scaling_instances({instance_ids: [instance_id]})
+                      .auto_scaling_instances[0]
+
+      new(
+        group_name: instance_info.auto_scaling_group_name,
+        id: instance_id,
+        region: region
+      )
+    end
+    
+    def terminate(options = {})
+      hook_name = options.delete(:continue_hook_name)
+      raw.terminate(options)
+      if hook_name
+        sleep(terminate_wait_sleep_seconds) until terminating_wait?
```

Comment:
should this have a max_attempts guard?  (in case, for whatever reason, the state gets stuck in terminating wait?

Reply:
considered that, but once a call to terminate happens, the logic doesn't want to do anything else, so not sure what we'd do once a max_attempts guard was triggered.  The lifecycle hooks will have their own timeout, and for the workers we will add a failsafe `shutdown` call on a timer.